File: historyscan-back/src/main/java/io/sch/historyscan/infrastructure/features/analysis/CodebaseToAnalyzeMapper.java
package io.sch.historyscan.infrastructure.features.analysis;

import io.sch.historyscan.domain.contexts.analysis.common.CodeBaseToAnalyze;
import org.springframework.stereotype.Component;

@Component
public class CodebaseToAnalyzeMapper {
    public CodeBaseToAnalyze webToDomain(CodeBaseToAnalyzeDTO codeBaseToAnalyzeDTO) {
        return CodeBaseToAnalyze.of(
                codeBaseToAnalyzeDTO.getName(),
                codeBaseToAnalyzeDTO.getType(),
                codeBaseToAnalyzeDTO.getRootFolder());
    }
}

File: historyscan-back/src/main/kotlin/io/sch/historyscan/infrastructure/features/analysis/CodebaseToAnalyzeMapper.kt
package io.sch.historyscan.infrastructure.features.analysis

import io.sch.historyscan.domain.contexts.analysis.common.CodeBaseToAnalyze
import io.sch.historyscan.infrastructure.features.analysis.dto.CodeBaseToAnalyzeDTO
import org.springframework.stereotype.Component

@Component
class CodebaseToAnalyzeMapper {
    fun webToDomain(codeBaseToAnalyzeDTO: CodeBaseToAnalyzeDTO): CodeBaseToAnalyze {
        return CodeBaseToAnalyze.of(
                codeBaseToAnalyzeDTO.name,
                codeBaseToAnalyzeDTO.type,
                codeBaseToAnalyzeDTO.rootFolder)
    }
}
